gen/generate.jl

```diff
@@ -0,0 +1,66 @@
+using Clang.cindex
+using Clang.wrap_c
+using Printf
+
+# Set these to correspond to your local filesystem's curl and clang include paths
+const CURL_PATH = "/usr/local/opt/curl/include/curl"
+const CLANG_INCLUDES = [
+    "/usr/local/opt/llvm/include",
+]
+
+const SRC_DIR = abspath(@__DIR__, "..", "src")
+
+headers = map(x-> joinpath(CURL_PATH, x), [ "curl.h", "easy.h", "multi.h"])
+context = wrap_c.init(;
+    headers = headers,
+    clang_args = String[],
+    common_file = joinpath(SRC_DIR, "lC_common_h.jl"),
+    clang_includes = CLANG_INCLUDES,
+    clang_diagnostics = true,
+    header_wrapped = (top_header, cursor_header) -> in(cursor_header, headers),
+    header_library = header -> "libcurl",
+    header_outputfile = header -> joinpath(SRC_DIR, "lC_") * replace(last(split(header, "/")), ".", "_")  * ".jl",
```

Comment:
Probably should be `replace(basename(header), ".", "_")`

Reply:
Or rather, `replace(basename(header), "." => "_")` for 0.7/1.0 support. This is available from Compat for 0.6.